### PR TITLE
Modis download return list of queryresults

### PR DESCRIPTION
**Pull request by interob**
See https://github.com/WFP-VAM/modape/issues/126

Introduced 2 return statements in the implementation of modis_download. Regarding re-use (chaining) of the vital "modis_???"  scripts, the use of sys.exit() is pretty evil any way... (fortunately, it was only in modis_download)

I propose renaming the --return-results flag into --print-results (not planning to use it, by the way)

(re-created against dev for https://github.com/WFP-VAM/modape/pull/127)

### CODE REVIEW

**Comment by valpesendorfer** [comment]
Few points:

- renaming of the flag is OK, definitely more clear
- Exiting a commandline script with `sys.exit` is pretty standard
- The returned results would be a list of dictionaries, each with format:
```python
{ 'file_id': '<modis_fid>',
  'time_start': '<start as datetime.date object>',
  'time_end': '<end as datetime object>',
  'updated': '<processing timestamp>',
  'link': '<download url>',
  'tile': '<tile identifier if applicable, otherwise None>'}
```

**Comment by interob** [comment]
Regarding the `sys.exit`: definitely pretty standard. But it complicates reuse of the script (calling it from from elsewhere). This PR reworks the `sys.exit` into a `return`

Returning a list of dicts is indeed what I currently work with.

File: modape/scripts/modis_download.py [hunk 1]
```diff
@@ -145,11 +146,11 @@ def cli(products: List[str],
 
     if query.nresults == 0:
         click.echo("No results found! Please check query or make sure CMR is available / reachable.")
-        sys.exit(0)
+        return []
```
Comment (valpesendorfer) [inline modape/scripts/modis_download.py:149]: can we do `return None`?

**Comment by interob** [comment]
Sure! Would that be more Pythonic? In my view, an empty list would nicely reflect nresults being zero

**Comment by valpesendorfer** [comment]
> Sure! Would that be more Pythonic? In my view, an empty list would nicely reflect nresults being zero

Actually, returning an empty list is probably more pythonic. I guess I'm still getting warmed up to the idea of returning a list from the cli function :D

File: modape/scripts/modis_download.py [hunk 2]
```diff
@@ -40,13 +40,13 @@ def cli(products: List[str],
...
         strict_dates: bool,
-        return_results: bool,
+        print_results: bool,
         download: bool,
         overwrite: bool,
         multithread: bool,
         nthreads: int,
         collection: str,
-        ) -> None:
+        ) -> list:
```
Comment (valpesendorfer) [inline modape/scripts/modis_download.py:49]: it should be `List` (as in `typing.List`)

**Review by valpesendorfer — CHANGES_REQUESTED** [review]

**Review by valpesendorfer — APPROVED** [review]

File: modape/scripts/modis_download.py [hunk 3]
```diff
@@ -72,6 +72,7 @@ def cli(products: List[str],
         multithread (bool): Use multiple threads for downloading.
         nthreads (int): Number of threads for multithread.
         collection (str): MODIS collection version.
+        :rtype: list
```
Comment (valpesendorfer) [inline modape/scripts/modis_download.py:75]: Sorry, one more thing I've realized:

For the docs, the return should be following the google schema:
```python

Returns:
        List of results returned by CMR API for query


```

**Review by valpesendorfer — CHANGES_REQUESTED** [review]

**Comment by interob** [comment]
I have hit the "Resolve conversation" button; that does the job?

**Comment by valpesendorfer** [comment]
> I have hit the "Resolve conversation" button; that does the job?

No sorry, this a new request I just saw before merging. It's just for the docstring

**Comment by interob** [comment]
So what is it you need from me?

**Comment by valpesendorfer** [comment]
> So what is it you need from me?

https://github.com/WFP-VAM/modape/pull/135#discussion_r539667222

File: modape/scripts/modis_download.py (hunk 3 above)
Comment (interob) [reply modape/scripts/modis_download.py:75]: Do I need to copy-paste that in my code editor? I am looking for an "Accept" button ;-)

Comment (valpesendorfer) [reply modape/scripts/modis_download.py:75]: Lol, yeah sorry. Copy and paste. But I give you the creative liberty to re-phrase. 

I would need to clone the branch and then add and push. I figured if you add it it's easier

Comment (interob) [reply modape/scripts/modis_download.py]: Well, I copy/pasted from PR thread into modis_download.py

Comment (interob) [reply modape/scripts/modis_download.py]: > Lol, yeah sorry. Copy and paste. But I give you the creative liberty to re-phrase.
> 
> I would need to clone the branch and then add and push. I figured if you add it it's easier

Sure. I just did not expect it was going to be an archaic workflow as copy/pasting

Comment (valpesendorfer) [reply modape/scripts/modis_download.py]: Sorry, I don't want to nitpick, but I didn't come up with the standards - and the automatic docbuilder might be failing.

Can you add a linebreak between the args and `Returns`?

Here's for copy-paste: :wink:  

```python
Args:
        products (List[str]): List of MODIS product codes to download.
        begin_date (datetime.date): Start date for query / download.
        end_date (datetime.date): End date for query / download.
        targetdir (pathlib.Path): Target directory.
        roi (str): Region of interest (point or bbox as csv string).
        tile_filter (str): MODIS tile filter (as csv string of tile IDs).
        username (str): Earthdata username.
        password (str): Earthdata password.
        strict_dates (bool): Strict date handling.
        download (bool): Download data.
        overwrite (bool): Replace existing.
        multithread (bool): Use multiple threads for downloading.
        nthreads (int): Number of threads for multithread.
        collection (str): MODIS collection version.
    
Returns:
        List of results returned by CMR API for query
```

**Review by valpesendorfer — APPROVED** [review]

**Comment by valpesendorfer** [comment]
Thanks @interob!

**Pull request merged**